backend/routing/QuizRoutes.go

```diff
@@ -0,0 +1,82 @@
+package routing
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/fourtf/studyhub/utils"
+)
+
+type apiResponse struct {
+	Error string      `json:"error"`
+	Data  interface{} `json:"data"`
+}
+
+type quiz struct {
```

Comment:
You can split up this struct so you don't need the MakeStrPtr function anymore. Just create the following structs:

- Question (consists of ID, Question)  //you might be more creative with the name of this struct or its variables than I am
- MultipleChoice (consists of Question, MultipleChoiceType(single, multiple) and Answers)
- FreeText (consists of Question and Answer)

Now you should be able to assign an array of questions to your data